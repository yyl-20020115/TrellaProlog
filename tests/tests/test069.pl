task70(X,Y) :-
	write('Frozen X='),
	write(X), Y=456,
	write(', set Y='),
	writeln(Y).

test70 :-
	freeze(X, task70(X,Y)),
	X=123, write('Y='),
	writeln(Y),
	writeln('OK done').

task71(X) :-
	write('Frozen X='),
	writeln(X), fail.

test71 :-
	freeze(X, task71(X)),
	X=123,
	writeln('Ooops').
test71 :-
	writeln('OK done').

task72(X) :-
	write('Frozen X='),
	writeln(X).

test72 :-
	X=123,
	freeze(X, task72(X)),
	writeln('OK done').

main :- test70, test71, test72, halt.

:- initialization(main).
