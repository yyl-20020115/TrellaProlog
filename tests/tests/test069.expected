Frozen X=123, set Y=456
Y=456
OK done
Frozen X=123
OK done
Frozen X=123
OK done
